internal/services/keyvault/key_vault_certificate_resource.go

```diff
@@ -681,15 +688,7 @@ func resourceKeyVaultCertificateDelete(d *pluginsdk.ResourceData, meta interface
 		return err
 	}
 
-	keyVaultIdRaw, err := keyVaultsClient.KeyVaultIDFromBaseUrl(ctx, resourcesClient, id.KeyVaultBaseUrl)
-	if err != nil {
-		return fmt.Errorf("retrieving the Resource ID the Key Vault at URL %q: %s", id.KeyVaultBaseUrl, err)
-	}
-	if keyVaultIdRaw == nil {
-		return fmt.Errorf("Unable to determine the Resource ID for the Key Vault at URL %q", id.KeyVaultBaseUrl)
-	}
-
-	keyVaultId, err := parse.VaultID(*keyVaultIdRaw)
+	keyVaultId, err := parse.VaultID(d.Get("key_vault_id").(string))
```

Comment:
unfortunately we can't use schema fields in the delete function - this isn't always available in the Plugin SDK and isn't available in Framework so we have to look this up from the API instead